Added BC817/BC807

eto- opened the pull request:
##### SUMMARY
Added BC817/BC807 complementary sot23 transistors


##### DATASHEETS / REFERENCES
https://www.mouser.it/datasheet/3/175/1/BC807-16_-25_-40.pdf
https://www.mouser.it/datasheet/3/175/1/ds11107.pdf


##### OPEN QUESTIONS / UNRESOLVED ISSUES
I added the models available on Mouser for the most common producers.
If these devices are fine, I can add others sot23 transistors.

##### CHECKLIST

- [x] I have read and followed the library conventions¹.
- [ ] For packages, I followed IPC7351C (see details in library conventions).
- [ ] For modified elements, I bumped their version number².
- [x] I already used the new library elements in a design and verified their correctness (optional).
- [x] I'm the copyright owner of the added content (i.e. the changes are made by myself, not copied/imported from somewhere else).
- [x] I agree to publish all my changes under the CC0 Public Domain License³, allowing everyone to use and modify the content without any restrictions.

¹ *Library Conventions: https://docs.librepcb.org/#libraryconventions*
² *Minor version bump if only metadata was modified (e.g. "0.1" -> "0.1.1"), major version bump if functional changes were made (e.g. "0.1" -> "0.2")*
³ *CC0 Public Domain License: https://en.wikipedia.org/wiki/CC0*

ubruhin commented:
Great choice to make a small PR, not all at once :smiley: 

I didn't review the MPNs exactly yet. But I have some feedback for the rest:

- The PNP devices uses the wrong component (NPN instead of PNP).
- The package looks wrong. According your linked datasheets, it should be "SOT23-3P95_230X110L50X40" (always be careful between "SOT23" and "SOT23-3" which are not the same).
- The manufacturer names are not consistent with other parts. They should use original capitalization as used on their own websites, Digikey etc, respectively consistent with the existing parts in our libraries. Those are:
  - "onsemi" (that one is fine as-is)
  - "diodes inc." -> "Diodes Incorporated"
  - "nexperia" -> "Nexperia"
- I think it would be nice to add a datasheet URL to the devices. Even though they wouldn't be valid for all the different MPNs, I'd say in this case we can allow that since those devices are considered generic anyway. In future we can even add multiple datasheets for the different manufacturers, it's just not yet supported by the UI.

eto- commented:
> * The PNP device uses the wrong component (NPN instead of PNP).

I was sure I changed that ... fixed

> * The package looks wrong. According your linked datasheets, it should be "SOT23-3P95_230X110L50X40" (always be careful between "SOT23" and "SOT23-3" which are not the same).

Yes, I'm not able to select it right now:
<img width="805" height="432" alt="image" src="https://github.com/user-attachments/assets/a0caa4e4-4c68-4f7d-b975-b9b4c740b062" />

Moreover there is confusion in the keywords: for SOT23 (the smaller) there is a "sot23-3" keyword and for the SOT23-3 (the bigger) there is a "sot-23" keyword. 

> * The manufacturer names are not consistent with other parts. They should use original capitalization as used on their own websites, Digikey etc, respectively consistent with the existing parts in our libraries. Those are:

Fixed

ubruhin commented:
> Yes, I'm not able to select it right now:

Hmm that is strange. But you see both SOT23 packages in the library?

> Moreover there is confusion in the keywords: for SOT23 (the smaller) there is a "sot23-3" keyword and for the SOT23-3 (the bigger) there is a "sot-23" keyword.

That is by intention. The terms "SOT23" and "SOT23-3" are sometimes not used correctly (IIRC even in datasheets it is sometimes misleading). By having both keywords in both packages, the user will always find both of them when searching for sot23, so he has to explicitly choose the correct one. If we didn't have those keywords, you can search for "sot23-3" and you will just take the single package which was found, even though it might be the wrong one.

ubruhin commented:
I have no idea why you don't see both SOT23 packages, while both are visible in the library. I just checked and for me it is there:

<img width="530" height="305" alt="image" src="https://github.com/user-attachments/assets/0b79a6f6-eddf-4b78-a9a5-235e9934c494" />

Would be interesting to understand how this can happen. If you are willing to investigate:
- Maybe there's any interesting logging output when running librepcb from the terminal
- Maybe the base library (or specifically that package) exists multiple times in your workspace (e.g. as a local library and as a remote library). And maybe one of those don't have the "SOT" category assigned for some reason, which would explain that behavior.

eto- commented:
I found that there is some sort of conflict with a local library containing devices not yet committed.
I will try to debug later. 

In the meantime I fixed the packages for the two transistors.

ubruhin commented:
Alright, I checked the MPNs, did some reordering and added a few more parts to BC807. I think now it looks good to me.

eto- commented:
> * Maybe the base library (or specifically that package) exists multiple times in your workspace (e.g. as a local library and as a remote library). And maybe one of those don't have the "SOT" category assigned for some reason, which would explain that behavior.

Indeed there was another package with the same UUID and funny it was a [sot variant](https://www.infineon.com/assets/row/public/documents/24/49/infineon-bga614-ds-en.pdf?fileId=db3a304314dca3890115418ffd35163a)

It was done when I was fresh new on librecad and probably I made a mistake.

I did not see any warning or error at the beginning: maybe we should add a warning if there are replicated UUID. I can write a small script to search replicated UUID if needed.

ubruhin commented:
Ok thank you for checking, then it makes sense this happened.

> I did not see any warning or error at the beginning: maybe we should add a warning if there are replicated UUID.

The library editor does show a big warning when you change the package or component of a device. But after it has been ignored once, there is no more warning afterwards. It would indeed be nice to detect & report this kind of problem, I opened https://github.com/LibrePCB/LibrePCB/issues/1620 to keep this in mind.
